docs/en/guides/pipeline-integrations/development/sgtk-developer-app.md

```diff
@@ -0,0 +1,257 @@
+---
+layout: default
+title: Developing apps
+pagename: sgtk-developer-app
+lang: en
+---
+
+# Developing your own app
+
+## Introduction
+
+This guide outlines what a Toolkit app is, and covers how to create your own, and also explains some of the basics around app development.
+Shotgun Pipeline Toolkit is not only a collection of apps and engines maintained by Shotgun Software, it is also a development platform that makes it easy to quickly build custom pipeline tools!
+
+- [What is a Toolkit app?](#what-is-a-toolkit-app)
+- [Creating your own app](#creating-your-own-app)
+
+Steps:
+1. [Creating a development sandbox](#part-1---creating-a-development-sandbox)
+2. [Forking or downloading the starter app repository](#part-2---forking-or-downloading-the-starter-app-repository)
+3. [Adding the app to your config](#part-3---adding-the-app-to-your-config)
+4. [Developing the app](#part-4---developing-the-app)
+    - [Anatomy of the Template Starter App](#anatomy-of-the-template-starter-app)
+    - [Configuration settings](#configuration-settings)
+    - [Frameworks](#frameworks)
+    - [Reloading your changes](#reloading-your-changes)
+5. [Testing](#part-5---testing)
+6. [Preparing your first release](#part-6---preparing-your-first-release)
+
+Additional info:
+- [Modifying an existing app](#modifying-an-existing-app)
+    - [Pull requests](#pull-requests)
+
+## What is a Toolkit app?
+
+A Toolkit app can be defined as follows:
+
+- A Tool that is usually run by an end-user, using the Shotgun integrations.
+- They usually have a graphical user interface with which the user can guide the app's operations, but it doesn't have to, it can also be a command registered with the integrations, triggered by the user from the Shotgun menu in the software.
+- They can have an API/public methods in which other processes or apps could interact.
+- They can be multi-platform, and software agnostic.
+- It can be set up to be configured differently per [environment](https://developer.shotgunsoftware.com/487a9f2c/?title=Environment+Configuration+Reference#what-is-an-environment).
+- They can be contextually aware. In other words can know the task the user is currently working in and act accordingly.
+- They can only be run from a Toolkit engine.
+
+Toolkit apps are initialized by Toolkit engines. Engines are designed to run within a specific software environment, where they then provide an interface from which to run Toolkit apps. The engine abstracts away the complexity of needing to handle the various software environments from the app.
+This means the app only needs to focus on providing the functionality to fulfill its purpose and doesn't need to, for example, handle window parenting, keeping track of the user's context, or providing a short cut for launching itself.
```

Comment:
- shortcut is a single word
 - Do we want to link to the "What is an engine?" section from the engine dev guide here?

Reply:
I've linked to the Engines class in the API instead. I'm open to putting another sentence in mentioning the build an engine guide specifically, though I think the engine reference documentation is probably all you need here.